apps/prairielearn/src/server.js

```diff
@@ -479,6 +479,13 @@ module.exports.initExpress = function () {
   if (config.devMode) {
     app.use('/pl/dev_login', require('./pages/authLoginDev/authLoginDev'));
   }
+  app.use('/pl/logout', [
+    function (req, res, next) {
+      res.locals.navPage = 'logout';
+      next();
+    },
+    require('./pages/authLogout/authLogout'),
```

Comment:
This page reads from `res.locals.authn_provider_name`, which is set by the `authn` middleware. That's actually the reason that these routes are sequenced how they are.

https://github.com/PrairieLearn/PrairieLearn/blob/6164c9d57bac1fa688f1e2ecafc35812fc563cc8/apps/prairielearn/src/pages/authLogout/authLogout.js#L24-L25

Given that built-in Shib support is very much deprecated, I'd be comfortable removing that branch entirely.

Reply:
That's also consistent with how the other authentication services logout works with PL cookies. We don't, for example, log you out of Google when we remove the PL cookie. I'm okay with removing this condition and extra redirect.

What's the impact of making that change? For us in production or others that might use the legacy Shibboleth?

Reply:
Zero impact for us, we're using per-institution SAML auth in production instead of the single hardcoded Shib provider. I think Berkeley might be using this Shib code, let me check if they're ok with this change.